lib/game.js

```diff
@@ -23,10 +23,33 @@ canvas.addEventListener('keydown', function(key) {
   } else if (key.which === 39) {
     boat.moveBoatRight(boat, canvasWidth);
   } else if (key.which === 32) {
+    filterBullets();
+    addBullet(bullets);
+  }
+});
+
+function addBullet(bullets) {
+  if (bullets.length === 0) {
```

Comment:
only fires if bullets array is empty, which is filtered by activeBullets prior to shoot method